### PR TITLE
docs: add demo token inspection

### PR DESCRIPTION
**Pull request by MadCcc**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [x] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 💡 Background and solution

- DemoWrapper 加了一个开关来控制是否展示 token
image
- Demo 添加了 token 栏，hover 可高亮变化
image
- 在 demo 代码中手动维护 meta，用于导出供编辑器消费
image

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     -      |
| 🇨🇳 Chinese |     -      |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 8eb9e26</samp>

Added a feature to show and hide demo tokens in the demo wrapper and code previewer components. Demo tokens are special comments that define the props and values used in the demo code. This feature helps developers to inspect and customize the demo code more easily. Updated the button demo and the locale files to support this feature.

### <samp>🤖 Generated by Copilot at 8eb9e26</samp>

*  Add a toggle button for showing or hiding demo tokens in the `DemoWrapper` component ([link](https://github.com/ant-design/ant-design/pull/41752/files?diff=unified&w=0#diff-e250ab9968952b8b565c98a89135c6b333a4dfaa67406881265a78f047a21cf6L4-R11), [link](https://github.com/ant-design/ant-design/pull/41752/files?diff=unified&w=0#diff-e250ab9968952b8b565c98a89135c6b333a4dfaa67406881265a78f047a21cf6R19), [link](https://github.com/ant-design/ant-design/pull/41752/files?diff=unified&w=0#diff-e250ab9968952b8b565c98a89135c6b333a4dfaa67406881265a78f047a21cf6R33-R36), [link](https://github.com/ant-design/ant-design/pull/41752/files?diff=unified&w=0#diff-e250ab9968952b8b565c98a89135c6b333a4dfaa67406881265a78f047a21cf6L45-R61), [link](https://github.com/ant-design/ant-design/pull/41752/files?diff=unified&w=0#diff-e250ab9968952b8b565c98a89135c6b333a4dfaa67406881265a78f047a21cf6L76-R104))
*  Import icons and tooltip text for the toggle button from `@ant-design/icons` and `locales` files ([link](https://github.com/ant-design/ant-design/pull/41752/files?diff=unified&w=0#diff-e250ab9968952b8b565c98a89135c6b333a4dfaa67406881265a78f047a21cf6L4-R11), [link](https://github.com/ant-design/ant-design/pull/41752/files?diff=unified&w=0#diff-77dff52935aad07ad5a5a47203c59f4d583e74034fd4695d189bd0c632b86e4bR23-R24), [link](https://github.com/ant-design/ant-design/pull/41752/files?diff=unified&w=0#diff-b823a78cf4beb8e8ede2c847a6fdfab8b61870bf9deffb35ec6fb7133384e281R23-R24))
*  Add a feature to highlight demo tokens in the live demo and display token tags below the code preview in the `CodePreviewer` component ([link](https://github.com/ant-design/ant-design/pull/41752/files?diff=unified&w=0#diff-54c90701ed595205ea2cd4286b1cd0ee8b2f01f77a23088e0108feda7af2d65fL9-R9), [link](https://github.com/ant-design/ant-design/pull/41752/files?diff=unified&w=0#diff-54c90701ed595205ea2cd4286b1cd0ee8b2f01f77a23088e0108feda7af2d65fR31), [link](https://github.com/ant-design/ant-design/pull/41752/files?diff=unified&w=0#diff-54c90701ed595205ea2cd4286b1cd0ee8b2f01f77a23088e0108feda7af2d65fL108-R115), [link](https://github.com/ant-design/ant-design/pull/41752/files?diff=unified&w=0#diff-54c90701ed595205ea2cd4286b1cd0ee8b2f01f77a23088e0108feda7af2d65fR136-R137), [link](https://github.com/ant-design/ant-design/pull/41752/files?diff=unified&w=0#diff-54c90701ed595205ea2cd4286b1cd0ee8b2f01f77a23088e0108feda7af2d65fL387-R428))
*  Import `ConfigProvider`, `Divider`, `Tag` and `useSiteToken` from `antd` and `hooks` to support the token highlighting feature ([link](https://github.com/ant-design/ant-design/pull/41752/files?diff=unified&w=0#diff-54c90701ed595205ea2cd4286b1cd0ee8b2f01f77a23088e0108feda7af2d65fL9-R9))
*  Pass the `tokens` and `showToken` props from the `Previewer` component to the `CodePreviewer` component ([link](https://github.com/ant-design/ant-design/pull/41752/files?diff=unified&w=0#diff-d4c2b096e898ee55c192cd1d13952ec5c5be4c1ff563d0ccaba74d8555e19adaR10-R11))
*  Define demo tokens for the basic button example in the `basic.tsx` file using the `tokens` comment ([link](https://github.com/ant-design/ant-design/pull/41752/files?diff=unified&w=0#diff-21525389d523cfebbe85bbc0d2314d68040a17a68c9ca757bb31f30331872e60R1-R3))

### CODE REVIEW

**Comment by afc163** [comment]
是不是需要一个 copy code，来展示怎么设置这个 token。

**Comment by MadCcc** [comment]
> 是不是需要一个拷贝 token 代码，来展示怎么在代码里设置这个 token。
> 
> > 下面的表格里也可以有。

拷贝 token 代码放到编辑器里了

**Comment by afc163** [comment]
> 拷贝 token 代码放到编辑器里了

最好在每个透出 token 的地方**就地**有个说明或者代码提示，不然用户想不到去哪里找。

**Comment by MadCcc** [comment]
> 最好在每个透出 token 的地方**就地**有个说明或者代码提示，不然用户想不到去哪里找。

这个以后可能每个 demo 都有，并且显示的 token 不一样，原地贴个链接或者代码的话感觉会比较烟花缭乱。
在下面的表格处放一个 tip 告诉用户这个功能如何呢？

**Comment by MadCcc** [comment]
@afc163 加了 Popover 链接到定制主题文档。放代码可能有些冗长，这里主要还是介绍每个 demo 会被哪些 token 影响。

File: .dumi/theme/builtins/Previewer/CodePreviewer.tsx [hunk 1]
```diff
@@ -384,11 +406,56 @@ createRoot(document.getElementById('container')).render(<Demo />);
...
+                        {locale.howToUseToken}
+                        <Link
+                          to={getLocalizedPathname('/docs/react/customize-theme', lang === 'cn')}
+                        >
+                          {locale.customizeTheme}
+                        </Link>
+                      </span>
+                    }
+                  >
+                    <InfoCircleOutlined style={{ marginLeft: 8 }} />
```
Comment (afc163) [inline .dumi/theme/builtins/Previewer/CodePreviewer.tsx:443]:
```suggestion
                    ，<span>（<QuestionCircleOutlined /> 如何使用 token？）</span>
```

Comment (MadCcc) [reply .dumi/theme/builtins/Previewer/CodePreviewer.tsx:443]: 那直接放链接么

**Comment by afc163** [comment]
另外一些 hover、focus 或者弹层里才能看到的样式，通过 hover tag 来预览可能就看不到了效果。

**Comment by li-jia-nan** [comment]
Conflicting

**Review by AbdullahWins — APPROVED** [review]
good job!

**Pull request closed without merging**